ArchiSteamFarm/Core/Utilities.cs

```diff
@@ -319,6 +325,42 @@ public static class Utilities {
 			}
 		}
 
+		internal static bool IsWeakPassword(string password, ISet<string>? additionallyForbiddenPhrases = null) {
+			if (string.IsNullOrEmpty(password)) {
+				throw new ArgumentNullException(nameof(password));
+			}
+
+			HashSet<string> forbiddenPhrases = ForbiddenPasswordPhrases.ToHashSet(StringComparer.InvariantCultureIgnoreCase);
+
+			if (additionallyForbiddenPhrases != null) {
+				forbiddenPhrases.UnionWith(additionallyForbiddenPhrases);
+			}
+
+			int remainingCharacters = password.Length;
+
+			foreach (string forbiddenPhrase in forbiddenPhrases.Where(static phrase => !string.IsNullOrEmpty(phrase))) {
+				for (int index = password.IndexOf(forbiddenPhrase, StringComparison.InvariantCultureIgnoreCase); index >= 0; index = password.IndexOf(forbiddenPhrase, index, StringComparison.InvariantCultureIgnoreCase)) {
+					remainingCharacters -= forbiddenPhrase.Length - 1;
+				}
+
+				if (remainingCharacters < 10) {
```

Comment:
Move this check inside the loop, it makes sense to check if we still meet the condition after every subtraction so we can stop parsing early.